shell/renderer/api/atom_api_web_frame.cc

```diff
@@ -436,6 +477,102 @@ v8::Local<v8::Promise> ExecuteJavaScriptInIsolatedWorld(
   return handle;
 }
 
+v8::Local<v8::Value> ExecuteJavaScriptSync(gin_helper::Arguments* args,
+                                           v8::Local<v8::Value> window,
+                                           const base::string16& code) {
+  v8::Isolate* isolate = args->isolate();
+  bool has_user_gesture = false;
+  args->GetNext(&has_user_gesture);
+
+  // return v8::Null(isolate);
+
+  ScriptSyncExecutionCallback* sync_callback =
+      new ScriptSyncExecutionCallback(isolate);
+
+  GetRenderFrame(window)->GetWebFrame()->RequestExecuteScriptAndReturnValue(
```

Comment:
This method on `WebFrame` is deprecated: https://cs.chromium.org/chromium/src/third_party/blink/public/web/web_local_frame.h?l=347&rcl=879cf27d2d56f2ff63e2ba7041bfabdd2d96deb3

Reply:
This is what the async versions are using as well, there's currently no alternative as the suggested requestExecuteScriptInIsolatedWorld does not work for the main world

Reply:
Oh? Even if you pass `0`, which is the ID for the main world?

Reply:
Ah, I see, the API checks that this is > 0. OK, it seems like there's no non-deprecated way to do this; presumably one will be added before this one is removed. Hopefully 😅